piksi_tools/bootload.py

```diff
@@ -58,25 +58,44 @@ def __exit__(self, *args):
       self.stop()
 
   def stop(self):
+    """ Remove Bootloader instance callbacks from serial link. """
     self.stopped = True
     self.link.remove_callback(self._deprecated_callback, SBP_MSG_BOOTLOADER_HANDSHAKE_DEPRECATED)
     self.link.remove_callback(self._handshake_callback, SBP_MSG_BOOTLOADER_HANDSHAKE_DEVICE)
 
   def _deprecated_callback(self, sbp_msg):
-    if len(sbp_msg.payload)==1 and struct.unpack('B', sbp_msg.payload[0])==0:
+    """ Bootloader handshake for deprecated message ID. """
+    hs_device = MsgBootloaderHandshakeDeprecated(sbp_msg)
+    if len(hs_device.handshake)==1 and hs_device.handshake[0]==0:
       # == v0.1 of the bootloader, returns hardcoded version number 0.
       self.version = "v0.1"
     else:
       # > v0.1 of the bootloader, returns git commit string.
-      self.version = sbp_msg.payload[:]
+      self.version = ''.join([chr(i) for i in hs_device.handshake])
```

Comment:
If this `hs_device.handshake` ends up being empty for some reason, will `self.version = ''`. Is that ok?